src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue

```diff
@@ -1273,9 +1295,156 @@ export default {
 				this.showCheckout();
 			}
 		},
+		amountLeftOnLoan(loan) {
+			const loanFundraisingInfo = loan?.loanFundraisingInfo ?? { fundedAmount: 0, reservedAmount: 0 };
+			const { fundedAmount, reservedAmount } = loanFundraisingInfo;
+			return numeral(loan.loanAmount).subtract(fundedAmount).subtract(reservedAmount).value();
+		},
+		allocateLeftoverCredits(payload) {
+			const newPayload = payload;
+			let loanIdx = newPayload.loanIdx ? newPayload.loanIdx : 0;
+			// Allocate leftover credits to a loan that isn't already in the basket
+			// and make sure it needs at least the leftover amount we want to allocate
+			while (
+				this.itemsInBasket.indexOf(this.availableLoans.values[loanIdx].id) > -1
+				|| newPayload.lendAmount > this.amountLeftOnLoan(this.availableLoans.values[loanIdx])
+			) {
+				loanIdx += 1;
+				if (loanIdx === this.availableLoans.length - 1) {
+					break;
+				}
+			}
+			const loanId = this.availableLoans.values[loanIdx].id;
+			setLendAmount({
+				amount: newPayload.lendAmount,
+				apollo: this.apollo,
+				loanId
+			}).then(() => {
+				this.$emit('add-to-basket', { loanId, success: true });
+				this.leftoverCreditAllocationLoanId = loanId;
+				this.updateBasketState();
+			}).catch(e => {
+				logFormatter(`Failed to add loan with id ${loanId} to basket`, 'error');
+				let msg = 'There was a problem adding the loan to your basket';
+				switch (e[0].extensions.code) {
+					case 'reached_anonymous_basket_limit':
+						msg = e[0].message;
+						this.$showTipMsg(msg, 'error');
+						break;
+					case 'no_shares_added_regular_xb' || 'not_all_shared_added':
+						msg = e[0].message;
+						loanIdx += 1;
+						newPayload.loanIdx = loanIdx;
+						this.allocateLeftoverCredits(newPayload);
+						break;
+					default:
+						this.$showTipMsg(msg, 'error');
+				}
+			});
+		},
+		updateBasketItem(payload) {
+			this.$emit('updating-totals', true);
+			const { loanId } = payload;
+			setLendAmount({
+				amount: payload.lendAmount,
+				apollo: this.apollo,
+				loanId: payload.loanId,
+			}).then(() => {
+				this.$emit('add-to-basket', { loanId, success: true });
+				this.leftoverCreditAllocationLoanId = loanId;
+				this.updateBasketState();
+				if (payload.lendAmount === Number(0)) {
+					this.completeRemoveBasketItem();
```

Comment:
I'm a bit concerned that this is a duplicate behavior. The actual removal of an item from basket and it's tracking will have been handled by the BasketItem component. Does this get called from some context outside of a BasketItem?

Reply:
@mcstover yes, it does.  When the basket is updated or when it is loaded, the CCLandingPage compares the amount of credit that is still available to the amount applied.  If there is leftover credit available to allocate towards a loan, the code chooses a loan from the carousel and applies that credit to it.  It also remembers in a cookie which loan had remaining credit allocated to it so that if a user adds more credit to a different loan, the credit is removed from that original loan that had credit allocated to it.  That removal process happens with no intervention from the user and occurs when the basket state is updated, so it wouldn't be something that originates from the "RemoveBasketItem" component in the BasketItem...

Reply:
Ok thank you, that makes sense. Just wanted to make sure we weren't calling the mutation or tracking the removal twice.